--- a/src/pluginscheduler.c	2019-07-12 15:52:15.551919519 +0300
+++ b/src/pluginscheduler.c	2019-07-12 15:58:47.487276103 +0300
@@ -305,7 +305,9 @@
       plugins_scheduler_free (ret);
       return NULL;
     }
+#ifdef __GLIBC__
   malloc_trim (0);
+#endif
   return ret;
 }
 
@@ -407,7 +409,9 @@
           element = element->next;
         }
     }
+#ifdef __GLIBC__
   malloc_trim (0);
+#endif
 }
 
 struct scheduler_plugin *
