--- a/base/hosts.c	2019-07-17 12:28:39.000000000 +0300
+++ b/base/hosts.c	2019-07-17 21:20:26.299147597 +0300
@@ -1013,7 +1013,9 @@
   hosts->count -= duplicates;
   hosts->removed += duplicates;
   hosts->current = 0;
+#ifdef __GLIBC__
   malloc_trim (0);
+#endif
 }
 
 /**
@@ -1208,7 +1210,9 @@
     gvm_hosts_deduplicate (hosts);
 
   g_strfreev (split);
+#ifdef __GLIBC__
   malloc_trim (0);
+#endif
   return hosts;
 }
 
